src/Evolution/Systems/ScalarAdvection/Subcell/NeighborPackagedData.hpp

```diff
@@ -0,0 +1,276 @@
+  apply(const db::DataBox<DbTagsList>& box,
+        const std::vector<std::pair<Direction<Dim>, ElementId<Dim>>>&
+            mortars_to_reconstruct_to) {
+    // The object to return
+    FixedHashMap<maximum_number_of_neighbors(Dim),
+                 std::pair<Direction<Dim>, ElementId<Dim>>, std::vector<double>,
+                 boost::hash<std::pair<Direction<Dim>, ElementId<Dim>>>>
+        neighbor_package_data{};
+
+    using evolved_vars_tags = typename System<Dim>::variables_tag::tags_list;
+    using fluxes_tags = typename Fluxes<Dim>::return_tags;
+
+    // subcell currently does not support moving mesh
+    ASSERT(not db::get<domain::Tags::MeshVelocity<Dim>>(box).has_value(),
+           "Haven't yet added support for moving mesh to DG-subcell. This "
+           "should be easy to generalize, but we will want to consider "
+           "storing the mesh velocity on the faces instead of "
+           "re-slicing/projecting.");
+
+    // Project volume variables from DG to subcell mesh
+    const Mesh<Dim>& dg_mesh = db::get<domain::Tags::Mesh<Dim>>(box);
+    const Mesh<Dim>& subcell_mesh =
+        db::get<evolution::dg::subcell::Tags::Mesh<Dim>>(box);
+    const auto volume_vars_subcell = evolution::dg::subcell::fd::project(
+        db::get<typename System<Dim>::variables_tag>(box), dg_mesh,
+        subcell_mesh.extents());
+
+    const auto& neighbor_subcell_data =
+        db::get<evolution::dg::subcell::Tags::
+                    NeighborDataForReconstructionAndRdmpTci<Dim>>(box);
+
+    const ScalarAdvection::fd::Reconstructor<Dim>& recons =
+        db::get<ScalarAdvection::fd::Tags::Reconstructor<Dim>>(box);
+
+    const auto& boundary_correction =
+        db::get<evolution::Tags::BoundaryCorrection<System<Dim>>>(box);
+    using derived_boundary_corrections =
+        typename std::decay_t<decltype(boundary_correction)>::creatable_classes;
+
+    // perform reconstruction
+    tmpl::for_each<derived_boundary_corrections>([&](auto
+                                                         derived_correction_v) {
+      using derived_correction =
+          tmpl::type_from<decltype(derived_correction_v)>;
+      if (typeid(boundary_correction) == typeid(derived_correction)) {
+        // U, NormalDotFlux<U>, AbsCharSpeed
+        using dg_package_field_tags =
+            typename derived_correction::dg_package_field_tags;
+        // VelocityField<Dim>
+        using dg_package_data_temporary_tags =
+            typename derived_correction::dg_package_data_temporary_tags;
+        // U, Flux<U>, VelocityField<Dim>
+        using dg_package_data_argument_tags =
+            tmpl::append<evolved_vars_tags, fluxes_tags,
+                         dg_package_data_temporary_tags>;
+
+        const auto& element = db::get<domain::Tags::Element<Dim>>(box);
+
+        // Variables to store packaged data
+        Variables<dg_package_field_tags> packaged_data{0};
```

Comment:
I think we can just default construct these two variables: `{}`. Let's do that. I realize GRMHD doesn't and you just copied it. That was poorly written code on my end, I think.